bin/index.js

```diff
@@ -1,71 +1,561 @@
+  }
+
+  let response = await requestFetch(requestOptions);
+  
+  if(rawResponseRaw) {
+    console.log(response);
+  } else {
+    console.log(response.data.result);
+  }
+}
+
+async function getLogsFunc(argv) {
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
+
+
+
+}
+
+async function pushFunc(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
 
-// Set up CLI
-const options = yargs
-  .usage('Usage: bsc -i <targetIp> -p <targetPassword> -m <reqMethod> -r <reqRoute>')
-  .usage('')
-  .usage('       Examples:')
-  .usage('       bsc -i=192.168.128.148 -p=ABC01A000001 -m=GET -r="/info"')
-  .usage('       bsc -i=192.168.128.148 -p=ABC01A000001 -m=GET -r="/files/sd"')
-  .option('i', { alias: 'targetIp', describe: 'IP Address of Target Player', type: 'string', demandOption: true })
-  .option('p', { alias: 'targetPassword', describe: 'Password of Target Player', type: 'string', demandOption: false })
-  .option('m', { alias: 'reqMethod', describe: 'Request method type', type: 'string', demandOption: true })
-  .option('r', { alias: 'reqRoutes', describe: 'Request url route', type: 'string', demandOption: true })
-  .option('a', { alias: 'raw', describe: 'Raw HTTP REST Response', type: 'boolean', demandOption: false })
-  // .option('s', { alias: 'dest', describe: 'Target Destination. e.g., sd/myfolder/', type: 'string', demandOption: false })
-  // .option('f', { alias: 'file', describe: 'File name or source file path to upload to target player', type: 'string', demandOption: true })
-  .argv;
-
-// CLI Arguments
-let ipAddress = options.i;
-let targetPassword = options.p;
-let requestMethod = options.m;
-let requestRoute = options.r;
-let rawResponse = options.a;
-
-// PUT e.g., /files/sd + file blob. 
-// let destDirectory = options.s;
-// let filePath = options.f;
-
-let axiosDigestAuthInst;
-
-async function prepareAxios () {
-    if (axiosDigestAuthInst == null || axiosDigestAuthInst === undefined) {
-        const options = {
-            username: 'admin',
-            password: targetPassword,
-        };
-        axiosDigestAuthInst = new AxiosDigestAuth(options);
+  let requestOptions = {
+    method: 'PUT',
+    url: 'http://' + playerIP + '/api/v1/files/sd/' + argv.location,
+  };
+
+  // check if file or directory
+  let path = argv.FileDirectory;
+  let absPath = currentPath.resolve(path);
+  let isFile;
+
+  isFile = await checkDir(path);
+  //console.log(isFile);
+
+  let files = [];
+  if(!isFile) {
+    console.log('getting files...');
+    try {
+      files = await getFiles(path);
+    }
+    catch (err) {
+      console.log('Error getting files from directory!');
+      console.log(err);
     }
-};
+  }
 
-async function requestAxios (requestOptions) {
-    let response;
+  if (isFile) {
```

Comment:
The logic within the if / else condition may benefit from a single try / catch block, rather than the individuals ones.